rclcpp_action/include/rclcpp_action/create_client.hpp

```diff
@@ -59,20 +59,19 @@ create_client(
         return;
       }
       auto shared_node = weak_node.lock();
-      if (!shared_node) {
```

Comment:
I personally prefer the early return as it uses less indentation on the code below and is easier to add to at a later date. With regard to coverage, I think it's also misleading because you might be increasing line coverage, but you're not actually testing the case where shared_node is null.

Instead of removing the early return, is there a way to test the case where the shared_pointer pointed to by the weak_pointer is deleted before this code gets run?

Reply:
The change is there to avoid leaking memory. I guess we could early return and `delete ptr` again. About coverage, it is being tested by https://github.com/ros2/rclcpp/pull/1153/files#diff-2ba27c7ee6e1481b18822aafc6b9eefaR298 and https://github.com/ros2/rclcpp/pull/1153/files#diff-2ba27c7ee6e1481b18822aafc6b9eefaR303.

Reply:
Oh shoot, my bad. I guess I missed that. I would leave it as you have it then.